File: CatsDemo_InsideOutTDD_DotNetCore/src/Interactors/CreateCatInteractor.cs
using Domain;
using Domain.Repositories;

namespace Interactors
{
    public class CreateCatInteractor
    {
        private ICatsRepository repository;
        
        public CreateCatInteractor(ICatsRepository repository)
        {
            this.repository = repository;
        }

        public Cat Execute(string name)
        {
            var cat = new Cat
            {
                Name = name
            };

            repository.Save(cat);

            return cat;
        }
    }
}

File: practice-2017-10-06-android/CleanArchitectureAndTDDDemo/core/src/main/java/com/example/interactors/createcat/CreateCatInteractor.java
package com.example.interactors.createcat;

import com.example.entities.Cat;
import com.example.repositories.CatsRepository;

import java.util.UUID;

/**
 * Created by MyndDev on 06/10/2017.
 */

public class CreateCatInteractor {

    CatCreatedPresenter presenter;
    private final CatsRepository repository;

    public CreateCatInteractor(CatCreatedPresenter presenter, CatsRepository repository) {
        this.presenter = presenter;
        this.repository = repository;
    }

    public void execute(CreateCatRequest request) {
        Cat cat = new Cat(UUID.randomUUID());
        cat.setName(request.name);
        cat.setTitle(request.title);
        repository.save(cat);

        CreateCatResponse response = new CreateCatResponse();
        response.id = cat.getId();
        response.name = request.name;
        response.title = request.title;

        presenter.present(response);
    }
}
